transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/DependencyConfiguration.java

```diff
@@ -5,24 +5,125 @@
  */
 package com.linkedin.transport.plugin;
 
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
 /**
  * Represents a dependency to be applied to a certain sourceset configuration (e.g. implementation, compileOnly, etc.)
  * In the future can expand to incorporate exclude rules, dependency substitutions, etc.
  */
 public class DependencyConfiguration {
   private ConfigurationType _configurationType;
-  private String _dependencyString;
+  private String _module;
+  private String _version;
+  private String _classifier;
+  private Set<PackageModule> _excludedPackageModules;
+
+  private DependencyConfiguration() {
+
+  }
+
+  public DependencyConfiguration(Builder builder) {
+    this._configurationType = builder._configurationType;
+    this._module = builder._module;
+    this._version = builder._version;
+    this._classifier = builder._classifier;
+    this._excludedPackageModules = builder._excludedPackageModules;
 
-  public DependencyConfiguration(ConfigurationType configurationType, String dependencyString) {
-    _configurationType = configurationType;
-    _dependencyString = dependencyString;
   }
 
   public ConfigurationType getConfigurationType() {
     return _configurationType;
   }
 
   public String getDependencyString() {
-    return _dependencyString;
+    return _module + ":" + _version + Optional.ofNullable(_classifier).map(v -> ":" + v).orElse("");
+  }
+
+  /**
+   * Return to the presentation of excluded packages
+   */
+  public Set<Map<String, String>> getExcludedPackages() {
+    return Optional.ofNullable(_excludedPackageModules)
+        .map(elements ->
+            elements.stream()
+                .filter(Objects::nonNull)
+                .map(PackageModule::toProperties)
+                .collect(Collectors.toSet()))
+        .orElse(null);
+  }
+
+  public Set<PackageModule> getExcludedPackageModules() {
+    return _excludedPackageModules;
+  }
+
+  public static DependencyConfiguration of(final ConfigurationType configurationType, final String module, final String version) {
+    return of(configurationType, module, version, null);
+  }
+
+  public static DependencyConfiguration of(final ConfigurationType configurationType, final String module, final String version, final String classifier) {
+    Objects.requireNonNull(configurationType);
+    Objects.requireNonNull(module);
+    Objects.requireNonNull(version);
+    return builder()
+        .configurationType(configurationType)
+        .module(module)
+        .version(version)
+        .classifier(classifier)
+        .build();
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
```

Comment:
So we have `Builder`, `of`, and constructor, and all are public. Can we just stick to a set of constructors?

Reply:
will use builder only in coming update